### PR TITLE
Mermaid: DirectedGraph to basic flowchart

### LINKED ISSUE #72: Add utilities to render `DirectedGraph` as Mermaid chart
## Example 1 - Mindmap

~~~mermaid
mindmap
Root
    A
      B
      C
~~~

## Example 2 - State Diagram

~~~mermaid
---
title: Simple sample
---
stateDiagram-v2
    [*] --> Still
    Still --> [*]

    Still --> Moving
    Moving --> Still
    Moving --> Crash
    Crash --> [*]
~~~

### PR DESCRIPTION
**Pull request by AustinArey**
Closes #72 

Thanks for the guidance coding this together in our Liveshare.

### CODE REVIEW

**Comment by AustinArey** [comment]
This PR has been ready to merge but unfortunately I left it open as I thought I might be able to add a feature; that would be belong in another PR. Ready!

File: mermaid/test.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Hello World!
```
Comment (pskfyi) [inline mermaid/test.md:1]: I think we don't want to commit this file.

File: mermaid/example.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { DirectedGraph } from "../graph/DirectedGraph.ts";
```
Comment (pskfyi) [inline mermaid/example.ts:1]: I think we don't want to commit this file, though parts of it should be relocated to a `readme.md` entry.

File: mermaid/flowchart.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { create as createFencedCodeblock } from "../md/codeBlock/fenced.ts";
+import { frontmatter } from "./_frontmatter.ts";
+//TODO accept object {to: , from: , text:}
```
Comment (pskfyi) [inline mermaid/flowchart.ts]: Created #100 to track 👍

File: readme.md [hunk 4]
```diff
@@ -247,6 +248,20 @@ codeBlock.evaluate(
 );
 ````
 
+## 'mermaid'
+
+```ts
+import { Mermaid } from "https://deno.land/x/handy/mermaid/utils.ts";
```
Comment (pskfyi) [inline readme.md:254]: `Mermaid` -> `flowchart`

File: readme.md [hunk 5]
```diff
@@ -247,6 +248,20 @@ codeBlock.evaluate(
...
+import { DirectedGraph } from "https://deno.land/x/handy/graph/utils.ts";
+
+const graph = new DirectedGraph<string>();
+
+graph.add(["a", "b", "c", "d"], ["h", "i", "j", "k"], ["c", "h"]);
+graph.add(["x", "y", "z"], ["z", "t"], ["z", "u"], ["d", "i"]);
+
+flowchart(graph.edges, { title: "Example Flowchart" }); //outputs mermaid flowchart markdown
```
Comment (pskfyi) [inline readme.md:262]: Can we show the generated flowchart in a real codeblock below? Would be 🔥.

File: readme.md [hunk 6]
```diff
@@ -14,6 +14,7 @@ Utility functions, classes, types, and scripts in uncompiled TS, for Deno.
 - [`io`](#io)
 - [`json`](#json)
 - [`md`](#md)
+- ['mermaid'](#mermaid)
   - [`script/evalCodeBlocks`](#scriptevalcodeblocks)
```
Comment (pskfyi) [inline readme.md:18]: Ordering issue. I recommend installing the Markdown All-In-One VS Code extension which automatically maintains this table of contents on save.

**Review by pskfyi — COMMENTED** [review]
Looking great!

A note on the commit message: Since this also initializes the `mermaid` module, I'd like to mention that in the message. Typically I just do `init; <other stuff`. So I'd recommend:

    feat(mermaid): init; add `flowchart()`

File: readme.md [hunk 7]
```diff
@@ -276,6 +277,35 @@ deno run --allow-read --allow-run \
   "some string to find" "replacement string" # optional
 ```
 
+## 'mermaid'
```
Comment (pskfyi) [inline readme.md:280]: These need to be backticks

Screenshot 2023-05-27 at 8 08 16 PM

File: readme.md [hunk 8]
```diff
@@ -276,6 +277,35 @@ deno run --allow-read --allow-run \
...
+flowchart LR
+    a --> b
+    b --> c
+    c --> d
+    c --> h
+    d --> i
+    h --> i
+    i --> j
+    j --> k
+```
```
Comment (pskfyi) [inline readme.md:307]: It looks fucking magical

File: readme.md (hunk 7 above)
Comment (AustinArey) [reply readme.md]: Thanks!

**Review by pskfyi — APPROVED** [review]
🧙 ⚡

**Pull request merged**